src/views/ProfilePage.vue

```diff
@@ -13,43 +17,142 @@
+                  <div v-if="userProfile?.friends?.length > 0">
+                    <ion-item-sliding
+                      v-for="(user, idx) in userProfile?.friends"
+                      :key="user?.id"
+                    >
+                      <ion-item>
+                        <div slot="start" v-if="user?.image">
+                          <user-avatar
+                            class="small avatar"
+                            :userProfile="user"
+                          />
+                        </div>
+                        <ion-label>{{ user?.name }}</ion-label>
+                      </ion-item>
+                      <ion-item-options>
+                        <ion-item-option
+                          color="danger"
+                          @click="removeUser(user.id)"
+                          >{{ $t("delete") }}</ion-item-option
+                        >
+                      </ion-item-options>
+                    </ion-item-sliding>
+                  </div>
+                  <div v-else>
+                    {{ $t("noFriends") }}
+                  </div>
+                </ion-list>
+                <friends-request-list :currentUserDoc="currentUserDoc" />
+                <ion-fab>
+                  <ion-fab-button id="modal-friends" expand="block">
+                    <ion-icon :icon="add"></ion-icon>
+                  </ion-fab-button>
+                </ion-fab>
+                <ion-modal ref="modal-friends" trigger="modal-friends">
+                  <ion-page>
+                    <ion-header>
+                      <ion-toolbar>
+                        <ion-buttons slot="start">
+                          <ion-button @click="cancel">Cancel</ion-button>
+                        </ion-buttons>
+                        <ion-title>{{ $t("addFriends") }}</ion-title>
+                      </ion-toolbar>
+                    </ion-header>
+                  </ion-page>
+                </ion-modal>
+              </ion-content>
+            </ion-page>
+          </ion-modal>
         </ion-item>
       </ion-list>
-      <ion-button class="logoutButton" expand="block" @click="logOut">Logout</ion-button>
+      <ion-button class="logoutButton" expand="block" @click="logOut"
+        >Logout
+      </ion-button>
     </ion-content>
   </ion-page>
 </template>
 
 <script setup>
-import { IonContent, IonPage, IonButton, IonSelect, IonSelectOption, IonItem, IonList, IonToggle, IonLabel } from '@ionic/vue';
-import { onBeforeMount, ref } from 'vue';
-import UserAvatarsList from '@/components/Base/UserAvatarsList.vue';
-import UserAvatar from '@/components/Base/UserAvatar.vue';
-import { useUserStore } from '@/plugins/pinia/users';
-import { useRouter } from 'vue-router';
+import {
+  IonContent,
+  IonPage,
+  IonButton,
+  IonIcon,
+  IonItem,
+  IonButtons,
+  IonTitle,
+  IonModal,
+  modalController,
+  IonToolbar,
+  IonHeader,
+  IonFabButton,
+  IonSearchbar,
```

Comment:
nitpick: Please add: ion-item-option, ion-item-options, ion-item-sliding, ion-list, ion-fab, ion-list-header, ion-label, ion-select, ion-select-option, ion-toggle to the imports !